### PR TITLE
Update metrics.py

### PR DESCRIPTION
**Pull request by LittlePea13**
Fix when less than 2 samples are selected for computing regression scores

### CODE REVIEW

**Comment by jameswex** [comment]
Thanks @LittlePea13 . What issue do you see when a single example is selected and regression metrics are calculated and displayed?

**Comment by LittlePea13** [comment]
It returns an error at the pearsonr calculation which shows up in the console and browser. I think a warning in the log this way may be better.

Right now it shows the following error:
```
E0908 16:19:48.566038 140495631603520 wsgi_app.py:208] Uncaught error: x and y must have length at least 2. 

 Traceback (most recent call last):
  File "/home/Documents/lit/lit_nlp/lib/wsgi_app.py", line 191, in __call__
    return self._ServeCustomHandler(request, clean_path)(environ,
  File "/home/Documents/lit/lit_nlp/lib/wsgi_app.py", line 176, in _ServeCustomHandler
    return self._handlers[clean_path](self, request)
  File "/home/Documents/lit/lit_nlp/app.py", line 77, in _handler
    outputs = fn(data, **kw)
  File "/home/Documents/lit/lit_nlp/app.py", line 240, in _get_interpretations
    config=data.get('config'))
  File "/home/Documents/lit/lit_nlp/api/components.py", line 71, in run_with_metadata
    model_outputs, config)
  File "/home/Documents/lit/lit_nlp/components/metrics.py", line 134, in run_with_metadata
    config=config.get(label_key) if config else None)
  File "/home/Documents/lit/lit_nlp/components/metrics.py", line 106, in compute_with_metadata
    return self.compute(labels, preds, label_spec, pred_spec, config)
  File "/home/Documents/lit/lit_nlp/components/metrics.py", line 173, in compute
    pearsonr = scipy_stats.pearsonr(labels, preds)[0]
  File "/home/anaconda3/envs/lit-nlp/lib/python3.7/site-packages/scipy/stats/stats.py", line 3501, in pearsonr
    raise ValueError('x and y must have length at least 2.')
ValueError: x and y must have length at least 2.

```

**Comment by jameswex** [comment]
Thanks @LittlePea13 . I would rather a line in the metrics table show up for the selection even when only a single point is selected, matching behavior in other demos and when more points are selected. When only a single example is selected, you could have the pearsonr value calculated at line 170 set to math.nan instead of calling into scipy. That will cause the display to show "-" in that column for the single selection, making it clear that it can't be calculated in that case. Also, I think we can skip the logging warning in this case as this is expected behavior when only selecting a single example.

**Comment by LittlePea13** [comment]
Thanks for the thoughts, I have updated the PR with what you suggest.

File: lit_nlp/components/metrics.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 from scipy import stats as scipy_stats
 from scipy.spatial import distance as scipy_distance
 from sklearn import metrics as sklearn_metrics
+import math
```
Comment (jameswex) [inline lit_nlp/components/metrics.py]: only need to import math if you want to use math.nan   
if using np.nan, as you are below, you don't need this import

File: lit_nlp/components/metrics.py [hunk 2]
```diff
@@ -166,12 +167,11 @@ def compute(self,
     if not labels or not preds:
       return {}
 
-    if len(labels)<2:
-      logging.warning("Number of selected items needs to be bigger than 2")
-      return {}
-
     mse = sklearn_metrics.mean_squared_error(labels, preds)
-    pearsonr = scipy_stats.pearsonr(labels, preds)[0]
+    if len(labels)<2: # Check if only one point selected.
```
Comment (jameswex) [inline lit_nlp/components/metrics.py:170]: nit: our style guide prefers a space before and after the "<" symbol

Comment (LittlePea13) [reply lit_nlp/components/metrics.py]: Noted, fixed it in a new commit.

**Review by jameswex — APPROVED** [review]

**Comment by jameswex** [comment]
Thanks @LittlePea13 !

**Pull request merged**